src/Parallel/PLMPI/CMakeLists.txt

```diff
@@ -6,17 +6,8 @@ add_library(PLMPI STATIC ${cppfiles})
 # Set common build options
 set_khiops_options(PLMPI)
 
-# findMPI fails on windows with clang, we add our own findMPI if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND
-# CMAKE_SYSTEM_NAME STREQUAL "Windows" ) if(NOT DEFINED ENV{MSMPI_INC}) message(FATAL "MSMPI is not installed") endif()
-# set(MPI_CXX_INCLUDE_PATH $ENV{MSMPI_INC}) set(MPI_CXX_LIBRARIES $ENV{MSMPI_LIB64}/msmpi.lib) #TODO test architecture
-# message("MPI is found at ${MPI_CXX_INCLUDE_PATH}") else() find_package(MPI 2.0 REQUIRED) endif() TODO ckeck MPI is
-# package is found
-target_include_directories(
-  PLMPI
-  PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}
-  PRIVATE ${MPI_CXX_INCLUDE_PATH})
-
+target_include_directories(PLMPI PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 target_link_libraries(
-  PLMPI PUBLIC PLParallelTask base) # We don't add ${MPI_CXX_LIBRARIES} here because we don't want that each binary that
-                                    # depends on PLMPI depends on libmpich.so (especially KNI)
+  PLMPI PUBLIC PLParallelTask base MPI::MPI_CXX) # By using the mpi target MPI_CXX, all is set correctly for mpi
```

Comment:
I don't understand what this exactly does. When adding `MPI::MPI_CXX` it will add the corresponding libraries if `FindMPI` was successful ? (I read a bit the documentation but is sparse). How it add the headers ? (they are added automatically if `FindMPI` is successful.

Reply:
Also. Does this solve #79 ?

Reply:
> Also. Does this solve #79 ?

I answer myself: No.

Reply:
Quand on ajoute une dépendance à une target, on prend ses propriétés, ici, les répertoire include, libs etc...
Il faut aller dans le script pour mieux comprendre https://github.com/Kitware/CMake/blob/master/Modules/FindMPI.cmake lignes 1219 1225 etc...

Reply:
Thanks!